tests/test_dimension.py

```diff
@@ -830,6 +831,98 @@ def test_implicit_dims(self):
 
         assert np.all(f.data == F)
 
+    def test_lowering(self):
+        """
+        Test the lowering of a ConditionalDimension's condition.
+        This test makes an Operator that should indexify and lower the condition
+        passed in the Conditional Dimension
+        """
+
+        grid = Grid(shape=(3, 3))  # Define grid
+
+        # Define Function
+        g = Function(name='g', grid=grid)
+        g.data[:] = 1
+        x, y = grid.dimensions
+        ci = ConditionalDimension(name='ci', parent=y, condition=Le(g, 1))
+
+        f = Function(name='f', shape=grid.shape, dimensions=(x, ci))
+        Operator(Eq(f, g)).apply()
+        assert np.all(f.data == 1)
+
+    def test_expr_like_lowering(self):
+        """
+        Test the lowering of an expr-like ConditionalDimension's condition.
+        This test makes an Operator that should indexify and lower the condition
+        passed in the Conditional Dimension
+        """
+
+        grid = Grid(shape=(3, 3))  # Define grid
+        # Define Function
+        g1 = Function(name='g1', grid=grid)
+        g2 = Function(name='g2', grid=grid)
+
+        g1.data[:] = 0.49
+        g2.data[:] = 0.49
+        x, y = grid.dimensions
+        ci = ConditionalDimension(name='ci', parent=y, condition=Le((g1 + g2),
+                                  1.01*(g1 + g2)))
+
+        f = Function(name='f', shape=grid.shape, dimensions=(x, ci))
+        Operator(Eq(f, g1+g2)).apply()
+
+        assert np.all(f.data[:] == g1.data[:] + g2.data[:])
+
+    @pytest.mark.parametrize('setup_rel, rhs, c1, c2, c3, c4', [
+        # Relation, RHS, c1 to c4 used as indexes in assert
+        (Lt, 3, 2, 4, 4, -1), (Le, 2, 2, 4, 4, -1), (Ge, 3, 4, 6, 1, 4),
+        (Gt, 2, 4, 6, 1, 4), (Ne, 5, 2, 6, 1, 2)
+    ])
+    def test_relational_classes(self, setup_rel, rhs, c1, c2, c3, c4):
```

Comment:
I actually think this and the two tests above could be squashed into a single test that tests lowering in the most general case , ie an expression `g1+g2` over a SubDomain. 

What do you think?

This three tests smells like over-engineering

Reply:
Unless I'm missing something, they all are testing, basically, that lowering _is happening_ -- not that "lowering works". So one general test should cover everything ?

Reply:
Yes, enhanced.